### PR TITLE
Added BitBar plugin

### PR DESCRIPTION
**Pull request by cj123**
[BitBar](https://github.com/matryer/bitbar) allows for simple plugins written with bash (etc).

I've put together this simple script to pull the information out of tid's status command, with resume and stop features for the current running timer.

image

This PR also creates an "extras" folder which completions and any other nice tid extras could be kept in.

### CODE REVIEW

**Comment by cj123** [comment]
Just noticed that this does not handle the case where there is no timer available to check the status of, when `tid status` returns "status: No timer to check the status of". (e.g. if you create and switch to a new workspace and then run the status command)

This actually leads to the statusbar showing "No", which isn't ideal 😄 .

I think making `tid status` exit with non-zero status in this case, and then detecting and handling it suitably in the plugin would be the best way of fixing this issue. Thoughts?

**Comment by seeruk** [comment]
Aah, that last part definitely needs changing when you're using `--format`, it should just output nothing in that case. I think you may have already covered the other half of that - setting error codes in those cases, so that scripts can react to no entries being found easier.

This looks really cool though, I wonder if when I get around to it if I switch from a DE to a WM with some kind of bar thing on Linux if I'll be able to make something similar for it...

Good shout on the extras directory too!

**Comment by n7st** [comment]
I've written a plugin in a fork [here](https://github.com/n7st/tid/tree/development/extras/i3blocks) for i3blocks (i3wm status bar), in case that's of any interest. It does need updating to read exit codes rather than using magic strings.

**Comment by seeruk** [comment]
Aha excellent. I'd only said to Callum the other day that it'd be great to have something for Linux (especially happy there because I've been considering the move to i3 for a while).

I'd love to see a PR!

File: pkg/tid/cli/command/entry/list.go [hunk 1]
```diff
@@ -85,6 +85,7 @@ func ListCommand(factory util.Factory) *console.Command {
 
 		if len(entries) == 0 {
 			output.Println("list: No entries within the given time period")
+			output.SetExitCode(1)
```
Comment (seeruk) [inline pkg/tid/cli/command/entry/list.go]: For these, I think we've already discussed, you can instead return an error to have it be a one-liner that sets the exit code too (though the addition of SetExitCode is still very useful).

File: pkg/tid/cli/command/entry/update.go [hunk 2]
```diff
@@ -50,6 +50,7 @@ func UpdateCommand(factory util.Factory) *console.Command {
 
 		if hasDuration && hasOffset {
 			output.Println("update: Duration and offset are mutually exclusive")
+			output.SetExitCode(1)
```
Comment (seeruk) [inline pkg/tid/cli/command/entry/update.go]: Same for the other instances of this.

File: glide.lock [hunk 3]
```diff
@@ -4,7 +4,7 @@ imports:
 - name: github.com/boltdb/bolt
   version: 583e8937c61f1af6513608ccc75c97b6abdf4ff9
 - name: github.com/eidolon/console
-  version: 9cf5800df5353e5ca5d9446c9c47b6921c9ad6cb
+  version: 1db5440ff756bcce3db592399d21b8f77e32b5f1
```
Comment (seeruk) [inline glide.lock:7]: I'm gonna check this now and see if we can release a new minor version.

**Review by seeruk — CHANGES_REQUESTED** [review]

File: glide.lock (hunk 3 above)
Comment (seeruk) [reply glide.lock:7]: v0.2.1 has been released, that should make this a bit safer.

**Review by seeruk — APPROVED** [review]

**Pull request merged**